static/js/publisher/market/market.js

```diff
@@ -88,32 +105,47 @@ function initForm(config, initialState, errors) {
   initSnapScreenshotsEdit(
     config.screenshotsToolbar,
     config.screenshotsWrapper,
-    state
+    state,
+    (nextState) => {
```

Comment:
Changes to screenshots need to trigger form validation, so we need to pass callback there.